pkg/dashboard/resource/project.go

```diff
@@ -372,6 +404,83 @@ func (pr *projectResource) importFunction(function *functionInfo, authConfig *pl
 	return nil
 }
 
+func (pr *projectResource) importProjectAPIGateways(projectImportInfoInstance *projectImportInfo,
+	failedFunctions []restful.Attributes) []restful.Attributes {
+
+	var failedAPIGateways []restful.Attributes
+
+	creationErrorContainsFunction := func(functionName string) bool {
+		for _, functionCreationError := range failedFunctions {
+			if functionCreationError["function"] == functionName {
+				return true
+			}
+		}
+		return false
+	}
+
+	// iterate over all api gateways and try to create each
+	for _, apiGateway := range projectImportInfoInstance.APIGateways {
+
+		// innocent until proven guilty
+		isValidAPIGateway := true
+
+		// validate upstreams
+		for _, upstream := range apiGateway.Spec.Upstreams {
+
+			// validate upstream kind
+			if upstream.Kind != platform.APIGatewayUpstreamKindNuclioFunction {
+				failedAPIGateways = append(failedAPIGateways, restful.Attributes{
+					"apiGateway": apiGateway.Spec.Name,
+					"error":      fmt.Sprintf("Unsupported api gateway upstream kind: %s", upstream.Kind),
+				})
+
+				isValidAPIGateway = false
+				break
+			}
+
+			// validate nucliofunciton spec is present
+			if upstream.Nucliofunction == nil {
+				failedAPIGateways = append(failedAPIGateways, restful.Attributes{
+					"apiGateway": apiGateway.Spec.Name,
+					"error":      "Upstream must contain nuclio function",
+				})
+
+				isValidAPIGateway = false
+				break
+			}
+
+			// validate upstream function is not one the functions that failed import
+			functionName := upstream.Nucliofunction.Name
+			if creationErrorContainsFunction(functionName) {
```

Comment:
After some thought I understood that this logic doesn't mean much, because if function is in error state we could still create api-gateway, fix the function and redeploy and already have the api gateway up and ready

Reply:
Yep, let's strive to keep the apigateway simple enough to not "carry" the state of the function